pkg/v1/tkg/client/management_components.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"os"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/managementcomponents"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgconfigreaderwriter"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/utils"
+)
+
+// InstallManagementComponents install management components to the cluster
+func (c *TkgClient) InstallManagementComponents(kubeconfig, kubecontext string) error {
+	managementPackageRepoImage, err := c.tkgBomClient.GetManagementPackageRepositoryImage()
+	if err != nil {
+		return errors.Wrap(err, "unable to get management package repository image")
+	}
+
+	// Override management package repository image if specified as part of below environment variable
+	mprImage := os.Getenv("MANAGEMENT_PACKAGE_REPO_IMAGE")
+	if mprImage != "" {
+		managementPackageRepoImage = mprImage
+	}
+
+	tkgPackageValuesFile, err := c.getTKGPackageConfigValuesFile()
+	if err != nil {
+		return err
+	}
+
+	managementcomponentsInstallOptions := managementcomponents.ManagementComponentsInstallOptions{
+		ClusterOptions: managementcomponents.ClusterOptions{
+			Kubeconfig:  kubeconfig,
+			Kubecontext: kubecontext,
+		},
+		KappControllerOptions: managementcomponents.KappControllerOptions{
+			KappControllerConfigFile:       "https://github.com/vmware-tanzu/carvel-kapp-controller/releases/download/v0.31.0/release.yml",
```

Comment:
Could we log an issue to clean this up to get the manifest from a template? kapp-controller has a bunch of configs specific to TKG.

Reply:
Added TODO and included issue https://github.com/vmware-tanzu/tanzu-framework/issues/1672